common/src/test/org/apache/hadoop/hive/common/TestFileUtils.java

```diff
@@ -338,4 +338,35 @@ private int assertExpectedFilePaths(RemoteIterator<? extends FileStatus> lfs, Li
     }
     return count;
   }
+
+  @Test
+  public void testResolveSymlinks() throws IOException {
+    HiveConf conf = new HiveConf();
+
+    java.nio.file.Path original = java.nio.file.Files.createTempFile("", "");
```

Comment:
can we have an additional case where there are nested links.
like 
symlink path -> symlink path -> actual path

means the provided symlink path resolves to a symlink rather than the actual path, we should get the final path ideally in this situation

Reply:
absolutely, I was thinking about that, but was lazy :)